url433bec1e56_sarobe_DifficultyInvestigation_tgz-pJ8-fr_inria_optimization_cmaes_examples_CMAExample1J8/src/fr/inria/optimization/cmaes/CMAEvolutionStrategy.java

```diff
@@ -0,0 +1,2704 @@
+          }
+          d[l] = d[l] + f;
+          e[l] = 0.0;
+       }
+      
+       // Sort eigenvalues and corresponding vectors.
+    
+       for (int i = 0; i < n-1; i++) {
+          int k = i;
+          double p = d[i];
+          for (int j = i+1; j < n; j++) {
+             if (d[j] < p) { // NH find smallest k>i
+                k = j;
+                p = d[j];
+             }
+          }
+          if (k != i) {
+             d[k] = d[i]; // swap k and i 
+             d[i] = p;   
+             for (int j = 0; j < n; j++) {
+                p = V[j][i];
+                V[j][i] = V[j][k];
+                V[j][k] = p;
+             }
+          }
+       }
+    } // tql2
+
+    /** not really in use so far, just clones and copies
+     * 
+     * @param popx genotype
+     * @param popy phenotype, repaired
+     * @return popy
+     */
+    double[][] genoPhenoTransformation(double[][] popx, double[][] popy) {
+    	if (popy == null || popy == popx || popy.length != popx.length) 
+    		popy = new double[popx.length][];
+    	
+    	for (int i = 0; i < popy.length; ++i)
+    		popy[i] = genoPhenoTransformation(popx[i], popy[i]);
+    	
+    	return popy;
+    }
+    /** not really in use so far, just clones and copies
+     * 
+     * @param popx genotype
+     * @param popy phenotype, repaired
+     * @return popy
+     */
+    double[][] phenoGenoTransformation(double[][] popx, double[][] popy) {
+    	if (popy == null || popy == popx || popy.length != popx.length) 
+    		popy = new double[popx.length][];
+    	
+    	for (int i = 0; i < popy.length; ++i)
+    		popy[i] = phenoGenoTransformation(popx[i], popy[i]);
+    	
+    	return popy;
+    }
+
+    /** not really in use so far, just clones and copies
+     * 
+     * @param x genotype
+     * @param y phenotype
+     * @return y
+     */
+    double[] genoPhenoTransformation(double[] x, double[] y) {
+    	if (y == null || y == x || y.length != x.length) {
+    		y = x.clone();
+    		return y; // for now return an identical copy
+    	}
+    	for(int i = 0; i < N; ++i)
+    		y[i] = x[i];
+    	return y;		
+    }
+    /** not really in use so far, just clones and copies
+     * 
+     * @param x genotype
+     * @param y phenotype
+     * @return y
+     */
+    double[] phenoGenoTransformation(double[] x, double[] y) {
+    	if (y == null || y == x || y.length != x.length) {
+    		y = x.clone();
+    		return y; // for now return an identical copy
+    	}
+    	for(int i = 0; i < N; ++i)
+    		y[i] = x[i];
+    	return y;		
+    }
+    
+    /**
+     * Samples the recent search distribution lambda times
+     * @return double[][] population, lambda times dimension array of sampled solutions, 
+     *   where <code>lambda == parameters.getPopulationSize()</code> 
+     * @see #resampleSingle(int)
+     * @see #updateDistribution(double[])
+     * @see CMAParameters#getPopulationSize()
+     */
+    public double[][] samplePopulation() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 19. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 1473-1481 into a separate method.